### ISSUE TITLE
"utils/int.h" referenced by ws.h, but not installed

### ISSUE THREAD

**Issue by djc** [nanomsg/nanomsg#467]
This seems wrong?

**Comment by wirebirdlabs** [comment]
tl;dr -- refactor away the original issue raised by @djc here in favor of per-socket rather than per-message set/get of RFC 6455 message type.

Further thoughts:

I consider this an as-yet unresolved issue brought up during the review of the WebSocket transport feature addition -- whether to expose WebSocket-specific message types defined in RFC6455 over the exported interface of libnanomsg, either thru the `cmsg` functions and/or thru websocket specific convenience wrappers.

Relevant code: https://github.com/nanomsg/nanomsg/blob/master/src/ws.h#L42-L60

The main issue, if I recall, is the classification of such message type as being either "transport level" or "application level" concern. A compelling argument was given that WebSocket attempted to wrap L6 concerns into its L4 protocol, and so the "bug"  lies within the design of RFC 6455. Any way this is argued -- it's still just an unresolved issue for libnanomsg.

I have a feeling that the reason this never came to a resolution is that it was hard to continue to scale the mindset of "all protocols across all transports, all obeying Liskov Substitution Principle" -- since there's definite value proposition in keeping WebSocket UTF-8 in order for libnanomsg to more easily communicate with browser clients.

That said -- I would probably favor refactoring the above-referenced lines to remove them, instead giving WebSocket a single transport level socket option that may be settable after constructing the socket -- Binary mode or UTF-8 mode. In practice, it's entirely reasonable to assume that any WebSocket endpoint would define an app-level messaging protocol that's strictly UTF-8 only or binary-only -- mixing/matching here feels like a terrible protocol design, and for this reason it would probably be OK to remove per-message control of message type.

**Comment by gdamore** [comment]
Concur.  The mode should be a socket option.  Its a shame that we have to
go thru these kind of hoops for JavaScript -- binary would be easier.  But
JS can't deal with real binary.
